src/Orleans/Streams/SimpleMessageStream/SimpleMessageStreamProducerExtension.cs

```diff
@@ -184,7 +191,7 @@ internal Task DeliverItem(StreamId streamId, object item, bool fireAndForgetDeli
             {
                 var tasks = fireAndForgetDelivery ? null : new List<Task>();
                 var immutableItem = new Immutable<object>(item);
-                foreach (KeyValuePair<GuidId, Tuple<IStreamConsumerExtension, IStreamFilterPredicateWrapper>> subscriptionKvp in consumers)
+                foreach (KeyValuePair<GuidId, Tuple<IStreamConsumerExtension, IStreamFilterPredicateWrapper>> subscriptionKvp in consumers.ToList())
```

Comment:
Now, that I think more, don't you have the same problem of concurrent collection modification in the pub sub grain? No, because of single threaded execution? But here, it can happen because it can run on the client?